crates/fuel-core/src/graphql_api/storage/coins.rs

```diff
@@ -2,27 +2,198 @@ use fuel_core_storage::{
 };
-use fuel_core_types::fuel_tx::{
-    Address,
-    TxId,
-    UtxoId,
+use fuel_core_types::{
+    entities::{
+        coins::coin::Coin,
+        Message,
+    },
+    fuel_tx::{
+        Address,
+        AssetId,
+        TxId,
+        UtxoId,
+    },
+    fuel_types::Nonce,
+};
+
+use crate::graphql_api::indexation;
+
+use self::indexation::coins_to_spend::{
+    NON_RETRYABLE_BYTE,
+    RETRYABLE_BYTE,
 };
 
+const AMOUNT_SIZE: usize = size_of::<u64>();
+const UTXO_ID_SIZE: usize = size_of::<UtxoId>();
+const RETRYABLE_FLAG_SIZE: usize = size_of::<u8>();
+
 // TODO: Reuse `fuel_vm::storage::double_key` macro.
 pub fn owner_coin_id_key(owner: &Address, coin_id: &UtxoId) -> OwnedCoinKey {
-    let mut default = [0u8; Address::LEN + TxId::LEN + 2];
+    let mut default = [0u8; Address::LEN + UTXO_ID_SIZE];
     default[0..Address::LEN].copy_from_slice(owner.as_ref());
-    let utxo_id_bytes: [u8; TxId::LEN + 2] = utxo_id_to_bytes(coin_id);
+    let utxo_id_bytes: [u8; UTXO_ID_SIZE] = utxo_id_to_bytes(coin_id);
     default[Address::LEN..].copy_from_slice(utxo_id_bytes.as_ref());
     default
 }
 
+/// The storage table for the index of coins to spend.
+
+// In the implementation of getters we use the explicit panic with the message (`expect`)
+// when the key is malformed (incorrect length). This is a bit of a code smell, but it's
+// consistent with how the `double_key!` macro works. We should consider refactoring this
+// in the future.
+pub struct CoinsToSpendIndex;
+
+impl Mappable for CoinsToSpendIndex {
+    type Key = Self::OwnedKey;
+    type OwnedKey = CoinsToSpendIndexKey;
+    type Value = Self::OwnedValue;
+    type OwnedValue = u8;
+}
+
+impl TableWithBlueprint for CoinsToSpendIndex {
+    type Blueprint = Plain<Raw, Primitive<1>>;
+    type Column = super::Column;
+
+    fn column() -> Self::Column {
+        Self::Column::CoinsToSpend
+    }
+}
+
+// For coins, the foreign key is the UtxoId (34 bytes).
+pub(crate) const COIN_FOREIGN_KEY_LEN: usize = UTXO_ID_SIZE;
+
+// For messages, the foreign key is the nonce (32 bytes).
+pub(crate) const MESSAGE_FOREIGN_KEY_LEN: usize = Nonce::LEN;
+
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct CoinsToSpendIndexKey(Vec<u8>);
+
+impl core::fmt::Display for CoinsToSpendIndexKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "owner={}, asset_id={}, retryable_flag={}, amount={}",
+            self.owner(),
+            self.asset_id(),
+            self.retryable_flag(),
+            self.amount()
+        )
+    }
+}
+
+impl CoinsToSpendIndexKey {
+    pub fn from_coin(coin: &Coin) -> Self {
+        let address_bytes = coin.owner.as_ref();
+        let asset_id_bytes = coin.asset_id.as_ref();
+        let retryable_flag_bytes = NON_RETRYABLE_BYTE;
+        let amount_bytes = coin.amount.to_be_bytes();
+        let utxo_id_bytes = utxo_id_to_bytes(&coin.utxo_id);
```

Comment:
As per the huddle, the `retryable_flag` will be moved to the very beginning of the key to allow queries like:
- give me all "retryable messages", regardless of the owner